src/contract_wrappers/exchange_wrapper.ts

```diff
@@ -583,14 +585,30 @@ export class ExchangeWrapper extends ContractWrapper {
         this._exchangeLogEventEmitters = [];
     }
     private _wrapEventEmitter(event: ContractEventObj): ContractEventEmitter {
+        const watch = (eventCallback: EventCallback) => {
+            const bignumberWrappingEventCallback = this._wrapEventCallback(eventCallback);
+            event.watch(bignumberWrappingEventCallback);
+        };
         const zeroExEvent = {
-            watch: event.watch.bind(event),
+            watch,
             stopWatchingAsync: async () => {
                 await promisify(event.stopWatching, event)();
             },
         };
         return zeroExEvent;
     }
+    private _wrapEventCallback(eventCallback: EventCallback): EventCallback {
+        const bignumberWrappingEventCallback = (err: Error, event: ContractEvent) => {
+            if (_.isNull(err)) {
+                const wrapIfBigNumber = (value: ContractEventArg): ContractEventArg => {
+                    return value instanceof String ? value : new BigNumber(value);
```

Comment:
Also, what if a value is returned that is too large to represent with a `number`. Wouldn't it be returned as a string?